vllm/v1/core/kv_cache_utils.py

```diff
@@ -124,6 +124,12 @@ class KVCacheBlock:
     prev_free_block: Optional["KVCacheBlock"] = None
     next_free_block: Optional["KVCacheBlock"] = None
 
+    # Whether the block is a null block that should never be cached.
+    # Use `is_not_null` instead of `is_null` because most checks in the
+    # codebase are for checking the block is not a null block,
+    # `block.is_not_null` is a little faster than `not block.is_null`
+    is_not_null: bool = True
```

Comment:
I honestly feel this is a bit over engineering 😅 if the performance is the only concern.

Reply:
OK. Changed to `is_null`.